pip_audit/_dependency_source/requirement.py

```diff
@@ -97,20 +98,26 @@ def collect(self) -> Iterator[Dependency]:
             # In order to get around this, we're going to copy each input into a
             # a corresponding temporary file and then pass that set of files
             # into `pip`.
-            tmp_files = []
-
-            # For each input file, copy it to one of our temporary files.
-            # Ensure we flush so our writes are visible to `pip`.
             for filename in self._filenames:
-                tmp_file = stack.enter_context(NamedTemporaryFile(mode="w"))
+                # Deliberately pass `delete=False` so that our temporary file doesn't get
+                # automatically deleted on close. We need to close it so that `pip` can
```

Comment:
I'm referring to `pip` here instead of the requirements parser (where the actual error is at the moment) since we're about to remove the parser from this code path.